src/rust/engine/process_execution/src/local.rs

```diff
@@ -335,12 +309,39 @@ impl CapturedWorkdir for CommandRunner {
     //
     // See: https://github.com/golang/go/issues/22315 for an excellent description of this generic
     // unix problem.
-    let _locked = if exclusive_spawn {
-      *self.spawn_lock.write().await
-    } else {
-      *self.spawn_lock.read().await
+    let mut child = {
+      let _locked = if exclusive_spawn {
+        *self.spawn_lock.write().await
+      } else {
+        *self.spawn_lock.read().await
+      };
+      command.spawn(Stdio::piped(), Stdio::piped())?
```

Comment:
This was a bug fix. PTAL. Previously locked synchronous creation of a future (child result stream) and not synchrnous execution of fork_exec via `tokio::process::spawn`.

Reply:
Erm, not a bad refactor, but also not true ... gimme a bit here.

Reply:
OK, I'm going to let this refactor stand even though it was not part of the actual DMM::contains_file fix if folks don't object. This is a step towards #10601 anyhow.